lib/ood_core/job/info.rb

```diff
@@ -111,6 +115,7 @@ def initialize(id:, status:, allocated_nodes: [], submit_host: nil,
         @status = job_array_aggregate_status unless @tasks.empty?
 
         @native          = native
+        @uses_gpu        = uses_gpu || native && native[:gres] ? native[:gres].include?("gpu") : false
```

Comment:
The calculation shouldn't be here because it will be adapter specific. You can add the `slurm` specifics in this PR - but basically when this object is built, the builder determines the flag and passes it in.

Also now -rethinking this, a job can request more than 1 gpu so - this should reflect that. I.e., the attribute is the number of gpus for the job ranging from 0 - N.

```ruby
   # the number of gpu's the job has requested
   attr_reader :gpus

   def initialize(
     # ...
     @gpus        =  gpus.to_i
   end
        
    def gpu?
       gpus.positive?
    end
```